federatedscope/cl/trainer/trainer.py

```diff
@@ -0,0 +1,42 @@
+from federatedscope.register import register_trainer
+from federatedscope.core.trainers import GeneralTorchTrainer
+from federatedscope.core.auxiliaries import utils
+import numpy as np
+
+
+class CLTrainer(GeneralTorchTrainer):
+    def _hook_on_batch_forward(self, ctx):
+        x, label = [utils.move_to(_, ctx.device) for _ in ctx.data_batch]
+#         print(len(x), x[0].size(), x[1].size(), label.size())
+        x1, x2 = x[0], x[1]
+        z1, z2 = ctx.model(x1, x2)
+        if len(label.size()) == 0:
+            label = label.unsqueeze(0)
+        ctx.loss_batch = ctx.criterion(z1, z2)
```

Comment:
`ctx.loss_batch`, `ctx.y_true` and `ctx.y_prob` won't be deleted after finishing the running routine and may cause memory leakage.
It is suggested to create them by `CtxVar(obj, lifecycle)` or manage the lifecycle by your self (delete them when they are not used)

Reply:
@xkxxfyf  please update according to dawei's comments.

Reply:
it updates